### PR TITLE
[SYCL] Fix event info queries for dummy non-host events

### PR DESCRIPTION
**Pull request by sergey-semenov**
It's possible to obtain a non-host event without an encapsulated
pi_event, e.g. as a result of a USM operation on 0 bytes. Handle this
case in event information queries.

### CODE REVIEW

**Comment by sergey-semenov** [comment]
Test: https://github.com/intel/llvm-test-suite/pull/201

**Review by s-kanaev — DISMISSED** [review]
LGTM

File: sycl/source/detail/event_impl.cpp [hunk 1]
```diff
@@ -226,8 +226,14 @@ template <>
...
-    return get_event_profiling_info<info::event_profiling::command_submit>::get(
-        this->getHandleRef(), this->getPlugin());
+    if (MEvent)
+      return get_event_profiling_info<
+          info::event_profiling::command_submit>::get(this->getHandleRef(),
+                                                      this->getPlugin());
+    else
+      // TODO this should throw an exception if the queue the dummy event is
+      // bound to does not support profiling info.
+      return 0;
```
Comment (bader) [inline sycl/source/detail/event_impl.cpp:236]:
```suggestion
    // TODO this should throw an exception if the queue the dummy event is
    // bound to does not support profiling info.
    return 0;
```

**Review by bader — COMMENTED** [review]
No else after return, please.

**Review by bader — APPROVED** [review]

**Review by s-kanaev — APPROVED** [review]

**Pull request merged**